### PR TITLE
Add warning about read-only invocation to cache card

### PR DESCRIPTION
**Pull request by maggie-lou**
Fixes buildbuddy-io/buildbuddy-internal#1346

### CODE REVIEW

File: app/invocation/invocation_model.tsx [hunk 1]
```diff
@@ -249,6 +250,15 @@ export default class InvocationModel {
     return this.invocations.find(() => true)?.acl?.groupId === "";
   }
 
+  isReadOnlyInvocation(): boolean {
```
Comment (bduffany) [inline app/invocation/invocation_model.tsx:253]: nit: consider renaming to something like `usedReadOnlyAPIKey()` or maybe more directly, `hasCacheWriteCapability()` and negate the return & callsites

**Review by bduffany — APPROVED** [review]
looks great!

File: app/invocation/invocation_cache_card.tsx [hunk 2]
```diff
@@ -12,21 +12,28 @@ const BITS_PER_BYTE = 8;
 
 export default class CacheCardComponent extends React.Component<Props> {
   render() {
+    const hasCacheStats =
+      this.props.model.cacheStats.length &&
+      (+this.props.model.cacheStats[0]?.actionCacheHits != 0 ||
+        +this.props.model.cacheStats[0]?.actionCacheMisses != 0 ||
+        +this.props.model.cacheStats[0]?.totalDownloadSizeBytes != 0 ||
+        +this.props.model.cacheStats[0]?.totalUploadSizeBytes != 0);
```
Comment (bduffany) [inline app/invocation/invocation_cache_card.tsx:20]: nit: prefer strict equality `!==` to loose equality `!=`

File: app/invocation/invocation_model.tsx (hunk 1 above)
Comment (maggie-lou) [reply app/invocation/invocation_model.tsx]: Done

File: app/invocation/invocation_cache_card.tsx (hunk 2 above)
Comment (maggie-lou) [reply app/invocation/invocation_cache_card.tsx]: Done

File: server/backends/invocationdb/invocationdb.go [hunk 3]
```diff
@@ -88,9 +90,18 @@ func (d *InvocationDB) registerInvocationAttempt(ctx context.Context, ti *tables
...
-		if u, err := auth.AuthenticatedUser(ctx); err == nil && u.GetGroupID() != "" {
-			permissions = perms.GroupAuthPermissions(u.GetGroupID())
+		u, err := auth.AuthenticatedUser(ctx)
+		if err == nil {
+			if u.GetGroupID() != "" {
+				permissions = perms.GroupAuthPermissions(u.GetGroupID())
+			}
+			caps = u.GetCapabilities()
+		} else {
+			if perms.IsAnonymousUserError(err) && auth.AnonymousUsageEnabled() {
```
Comment (maggie-lou) [inline server/backends/invocationdb/invocationdb.go:102]: @bduffany Realized I didn't handle anonymous users. Can you verify that it's the intended behavior here to use the default capabilities (includes write) for anonymous users?

Comment (bduffany) [reply server/backends/invocationdb/invocationdb.go:102]: Nice catch -- yep, this looks right.

**Review by bduffany — APPROVED** [review]

**Pull request merged**